extensions/Worker.Extensions.Http.AspNetCore/src/FunctionContext/AspNetExtensionHttpRequestDataFeature.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker.Http;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore
+{
+    internal class AspNetExtensionHttpRequestDataFeature : IHttpRequestDataFeature
+    {
+        private AspNetExtensionHttpRequestDataFeature()
+        {
+        }
+
+        public static AspNetExtensionHttpRequestDataFeature Instance { get; } = new AspNetExtensionHttpRequestDataFeature();
+
+        public ValueTask<HttpRequestData?> GetHttpRequestDataAsync(FunctionContext context)
+        {
+            throw new NotSupportedException($"The method {nameof(GetHttpRequestDataAsync)} " +
```

Comment:
Do we want to, or have any plans to, support gradual migration to this model? If we did add a shim support layer for this, then customers could enable this feature, keep their existing HTTP functions as is and migrate them over time.

Reply:
We've considered adding a support layer for this so that users can combine this method with the new model, but for the next preview we're going to throw an error for now. We'll use the error message in the logs to track how often people run into this and see what feedback we get. May add a shim support layer later - still up in the air.